### PR TITLE
[AM] [KD] Create New Shopping List

### LINKED ISSUE #4: 4. As a user, I want to set up a new shopping list so I can start tracking purchased items.
## Summary 

A shopping list is a set of items associated with the owner's unique id and the name of the shopping list. Users need to be able to create new shopping lists. To do this, the `Home` view should present them with a form that allows them to enter the name of their list. Additionally, we need to do a few things to create that list:

1. Save the token to `localStorage`
2. Save the list to our database using the `createList` function in `firebase.js` 
3. Show the user the list view
## Acceptance criteria
UI-related tasks:
- [x] The `Home` view displays a form that allows users to enter the name of a shopping list and then create a list with that name.
- [x] The input that accepts the name of the item has a semantic `label` element associated with it
- [x] The user can submit this form with both the mouse and the `Enter` key  
- [x] #20

Data-related tasks:
- [x] Clicking the button uses the `createList` function in `src/api/firebase.js` to create a new shopping list for the user.
- [x] The shopping list path is stored in local storage using the `setListPath` function.
- [x] Once the list has been created and saved, the user is redirected to the `List` view.

## Notes
- The `createList` function takes **three** arguments: the current user's id, the current user's email, and the name of the shopping list.
- The `setListPath` function takes **one** argument: the new list token.
- If you get stuck trying to redirect the user, check out the [useNavigate](https://reactrouter.com/en/6.14.2/hooks/use-navigate) hook from `react-rotuer`.

### PR DESCRIPTION
**Pull request by piecanoe**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._

## Description

This code allows user to create a new shopping list on Home and immediately redirect to List page. The new list is created on Firebase. 

Note: We used the 'useNavigate' hook instead of 'redirect' to redirect to the List view on submit. React router docs recommend using 'redirect' over 'useNavigate' although we had trouble implementing 'redirect'.

## Related Issue

Closes #4

## Acceptance criteria
UI-related tasks:
- [x] The `Home` view displays a form that allows users to enter the name of a shopping list and then create a list with that name.
- [x] The input that accepts the name of the item has a semantic `label` element associated with it
- [x] The user can submit this form with both the mouse and the `Enter` key  
- [x] When the user submits the form, they see a message indicating that the list either was or was not created and saved to the database. 

Data-related tasks:
- [x] Clicking the button uses the `createList` function in `src/api/firebase.js` to create a new shopping list for the user.
- [x] The shopping list path is stored in local storage using the `setListPath` function.
- [x] Once the list has been created and saved, the user is redirected to the `List` view.

## Type of Changes

Use one or more labels to help your team understand the nature of the change(s) you’re proposing. E.g., `bug fix` or `enhancement` are common ones.

### Before

issue4_before

### After

issue4_after

## Testing Steps / QA Criteria

Create a new list using the form. Make sure submit works on button click and using 'ENTER' key. An alert will notify you that a list has been create and immediately redirects to /list path. You can also check that the new list is created in the Firestore Database. Test that the app alerts an error if an empty string is submitted.

### CODE REVIEW

**Review by krsnamara — APPROVED** [review]
Great work yall! Can't wait to merge both and get my shopping lists started!

**Review by hsiangj — APPROVED** [review]
Works as expected, great work you two!

File: src/views/Home.jsx [hunk 1]
```diff
@@ -1,8 +1,29 @@
 import './Home.css';
+import { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
 import { SingleList } from '../components';
+import { createList } from '../api/firebase';
+
+export function Home({ data, setListPath, user }) {
+	const [shoppingListName, setShoppingListName] = useState('');
+	const navigate = useNavigate(); // useNavigate doc suggests using redirect
```
Comment (louieknolle) [inline src/views/Home.jsx:9]: According to some Googling, `redirect` is for use in route loaders and actions, not directly w/in components. So useNavigate is probably the best route (hehe) to go.

**Review by louieknolle — APPROVED** [review]
Looks good to me, works as expected and I wouldn't worry about the `redirect` from react-router-dom! (as noted in my comment in `Home.jsx`)

**Pull request merged**